src/components/fields/Percentage/Settings.tsx

```diff
@@ -0,0 +1,195 @@
+import { useState } from "react";
+
+import {
+  Box,
+  ButtonBase,
+  Checkbox,
+  Collapse,
+  InputLabel,
+  useTheme,
+} from "@mui/material";
+import ColorPickerInput from "@src/components/ColorPickerInput";
+import { ISettingsProps } from "@src/components/fields/types";
+
+import { Color, toColor } from "react-color-palette";
+import { fieldSx } from "@src/components/SideDrawer/utils";
+import { ChevronDown } from "mdi-material-ui";
+import { ReactElement } from "react-markdown/lib/react-markdown";
+import { resultColorsScale } from "@src/utils/color";
+
+const ColorPickerCollapse = ({
+  colorKey,
+  color,
+  active,
+  setActive,
+  disabled,
+  children,
+}: {
+  colorKey: string;
+  color: Color;
+  active: boolean;
+  setActive: (activeKey: string | null) => void;
+  disabled?: boolean;
+  children?: ReactElement;
+}) => {
+  const toggleCollapse = () => !disabled && setActive(active ? null : colorKey);
+
+  const toggleState = (() => {
+    if (disabled && active) {
+      setActive(null);
+      return false;
+    }
+    return active;
+  })();
+
+  return (
+    <Box sx={{ width: "100%" }}>
+      <ButtonBase
+        onClick={toggleCollapse}
+        component={ButtonBase}
+        focusRipple
+        disabled={disabled}
+        sx={[
+          fieldSx,
+          {
+            justifyContent: "flex-start",
+            "&&": { pl: 0.75, pr: 0.5 },
+            color: color.hex,
+            transition: (theme) =>
+              theme.transitions.create("border-radius", {
+                delay: theme.transitions.duration.standard,
+              }),
+            "&.Mui-disabled": { color: "text.disabled" },
+          },
+          active && {
+            transitionDelay: "0s",
+            transitionDuration: "0s",
+          },
+        ]}
+      >
+        <Box
```

Comment:
Again extract the wrapper, it isn't lovely and makes the logic seem more complex. Better to stay away from composition.